Readme draft, add tests & github publishing workflow (#1)
tl-its-umich-edu/react-ga-onetrust-consent: README.md

# react-ga-onetrust-consent

> React implementation of OneTrust consent banner integrated with Google Analytics signals. Initially built to support analytics-enabled University of Michigan Teaching & Learning applications, this logic using Google Analytics tracking alongside OneTrust consent verification is written to be properly standardized; We will aim to iterate on this project such that it could be shared with other UofM web apps that utilize React and Google Analytics. 

Reviewer: Rather than `UofM`, use `UMich` if there's any chance this module will be used by other organizations or universities.  Otherwise, `U-M` is commonly used in communications, although it is ambiguous, too.

```suggestion
React implementation of OneTrust consent banner integrated with Google Analytics signals. Initially built to support analytics-enabled University of Michigan Teaching & Learning applications, this logic using Google Analytics tracking alongside OneTrust consent verification is written to be properly standardized; We will aim to iterate on this project such that it could be shared with other UMich web apps that utilize React and Google Analytics. 
```

Author: Sure, makes sense to me.

For more information on the University of Michigan's technical implementation of OneTrust, see the OVPC Digital team's [instructions on cookie disclosures](https://vpcomm.umich.edu/resources/cookie-disclosure/).

## Table of Contents

- [Overview](#overview)
- [Installation](#installation)
- [Usage](#usage)
- [Development](#development)
- [Testing](#testing)
- [Releasing](#releasing)
- [License](#license)

# Overview

Applications should import and apply the hook `useGoogleAnalytics` at the root level of App in the DOM. Upon initializing with the proper settings, this hook will call upon the `useOneTrust` hook which initializes the consent banner and captures user preferences.

Beyond simply loading a script for the banner, the `useOneTrust` hook also writes a callback to properly emit settings changes to Google Analytics and delete Google Analytics cookies if tracking is denied (EU countries only). To get a better understanding how OneTrust and Google Analytics work together, see the [UofM Google Analytics implementation instructions](https://vpcomm.umich.edu/resources/cookie-disclosure/#3rd-party-google-analytics).

> Remember to keep in mind that the consetn banner h

Author: thought completed :)

# Installation

> You can install this package via GitHub Packages. Add the following configuration to your .npmrc file to use GitHub Packages registry:

Reviewer: Minor comment:  Don't use "you" in documentation.  That's been recommended by documentation writers in the past.  Remove it from your sentences and rewrite them to make sense without it.  Maybe…

```suggestion
This package may be installed via GitHub Packages.  Add the following configuration to the `.npmrc` file to use GitHub Packages registry:
```

Author: Good suggestions. I went ahead and kept the term "you"/"your" out of  the readme, let me know if it could be improved!

> You can install this package via GitHub Packages. Add the following configuration to your .npmrc file to use GitHub Packages registry:

Reviewer 2: where is this file `.npmrc` located? on local machine? where at the project level or Home directory ( for Macs)

Author: After discovering the alternative discussed in the dev meeting, I added a disclaimer in the readme that npmrc is no longer necessary.

Reviewer 2: Nice

`@tl-its-umich-edu:registry=https://npm.pkg.github.com //npm.pkg.github.com/:_authToken=YOUR_PERSONAL_ACCESS_TOKEN `

> 

Reviewer 2: Where should I get this `YOUR_PERSONAL_ACCESS_TOKEN`?

Author: Changed readme. personall access token is No longer necessary.
